pkg/controller/keystone/keystone_controller.go

```diff
@@ -296,6 +310,30 @@ func (r *ReconcileKeystone) updateStatus(
 	return r.client.Status().Update(context.Background(), k)
 }
 
+func (r *ReconcileKeystone) externalKeystoneReady(k *contrail.Keystone, adminPasswordSecret string) (bool, error) {
+	keystoneClient, err := keystone.NewClient(r.client, r.scheme, r.restConfig, k)
+	if err != nil {
+		return false, err
+	}
+	_, err = keystoneClient.PostAuthTokens("admin", adminPasswordSecret, "admin")
```

Comment:
maybe we can introduce some retry mechanism here?

Reply:
I don't have too much experience yet with operator, my current understanding is that in case of error reconcile will be repeated, which I saw as some kind of retry mechanism.

Reply:
You are right I was thinking that connection to some external endpoint can be more flaky than within the cluster so maybe we could check the POST request with a retry mechanism but ofc it's not a bug to return with error from Reconcile and let it be invoked later. That was an idea that came to my mind but it is not a necessary feature, it can stay like that.